### PR TITLE
Correction to UnitarySystem when HumRat control node is different from Temp control node

### LINKED ISSUE #7409: UnitarySystem water cooling coil using CoolReheat shows HumRatMax SPM not found
Issue overview
--------------
User was confused with message of missing set point manager when the manager was included. The UnitarySystem checks for the SPM at the system outlet node instead of the coil outlet node.

User got this message:

```
   ** Severe  ** AirLoopHVAC:UnitarySystem: Missing humidity ratio setpoint (HUMRATMAX) for unitary system = HVAC PACKAGE
   **   ~~~   **   use a Setpoint Manager to establish a setpoint at the coil control node.
```

and then added another SPM to try to resolve it. And then got:

```
   ** Warning ** SetpointManager:Scheduled="COIL MGR2"
   **   ~~~   ** ...setpoint node conflicts with another setpoint manager.
   **   ~~~   ** ...conflicting setpoint manager = SetpointManager:SingleZone:Humidity:Maximum:"ZONE MAX SET POINT MANAGER"
   **   ~~~   ** ...conflicting node name = FURNACE HEATING COIL 1 INLET
   **   ~~~   ** ...control type variable = MaximumHumidityRatio
   ** Severe  ** AirLoopHVAC:UnitarySystem: Missing humidity ratio setpoint (HUMRATMAX) for unitary system = HVAC PACKAGE
   **   ~~~   **   use a Setpoint Manager to establish a setpoint at the coil control node.
```

HumRatMax SPM is obviously present at the cooling coil outlet node (same as heating coil inlet node).

```
SetpointManager:Scheduled,
    Coil Mgr2,               !- Name
    MaximumHumidityRatio,    !- Control Variable
    Cooling Coil HumSetpoint,!- Schedule Name
    Furnace Heating Coil 1 Inlet;  !- Setpoint Node or NodeList Name

SetpointManager:SingleZone:Humidity:Maximum,
    Zone Max Set Point Manager,  !- Name
    Furnace Heating Coil 1 Inlet,  !- Setpoint Node or NodeList Name
    SalesFloor Node;         !- Control Zone Air Node Name
```

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (proprietary input file)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
Fixes #7409. Allows the HumRat set point manager to place a humidity control set point at a node other than the temperature control node.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

File: tst/EnergyPlus/unit/UnitarySystem.unit.cc [hunk 1]
```diff
@@ -2575,15 +2575,15 @@ TEST_F(ZoneUnitarySysTest, UnitarySystemModel_WaterCoilSPControl)
...
     // CW air inlet node temp is greater than CW air outlet node temp
     // TODO: FIXME: following is failing for some reason even after correcting nodes.
-    // EXPECT_GT( Node( coolingCoilAirInletNodeIndex ).Temp, Node( coolingCoilAirOutletNodeIndex ).Temp );
+    EXPECT_GT(DataLoopNode::Node(coolingCoilAirInletNodeIndex).Temp, DataLoopNode::Node(coolingCoilAirOutletNodeIndex).Temp);
     // CW water inlet node flow is greater than 0
     // TODO: FIXME: following is failing for some reason even after correcting nodes.
     // EXPECT_GT( Node( coolingCoilWaterInletNodeIndex ).MassFlowRate, 0.0 );
     // CW water node flow is the same at inlet and outlet
-    EXPECT_EQ(DataLoopNode::Node(coolingCoilWaterInletNodeIndex).MassFlowRate, DataLoopNode::Node(11).MassFlowRate);
+    EXPECT_NEAR(DataLoopNode::Node(coolingCoilWaterInletNodeIndex).MassFlowRate, DataLoopNode::Node(11).MassFlowRate, 0.0001);
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/UnitarySystem.unit.cc:2596]: Nice!

File: src/EnergyPlus/HVACInterfaceManager.cc [hunk 2]
```diff
@@ -691,12 +691,18 @@ namespace HVACInterfaceManager {
 
         } else { // tank has mass
             if (MassFlowRate > 0.0) {
-                TankFinalTemp = (LastTankOutletTemp - (MassFlowRate * Cp * TankInletTemp + PumpHeat) / (MassFlowRate * Cp)) *
-                                    std::exp(-(MassFlowRate * Cp) / (ThisTankMass * Cp) * TimeStepSeconds) +
+                // original code is: std::exp(-(MassFlowRate * Cp) / (ThisTankMass * Cp) * TimeStepSeconds)
+                Real64 stdexpMCp = (MassFlowRate * Cp) / (ThisTankMass * Cp) * TimeStepSeconds;
+                if (stdexpMCp < 700.0) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/HVACInterfaceManager.cc]: Adding protection here is fine.

File: src/EnergyPlus/UnitarySystem.cc [hunk 3]
```diff
@@ -1370,11 +1370,11 @@ namespace UnitarySystems {
             if ((this->m_DehumidControlType_Num != DehumCtrlType::None) &&
                 (DataLoopNode::Node(ControlNode).HumRatMax == DataLoopNode::SensedNodeFlagValue) && this->m_ControlType == ControlType::Setpoint &&
                 CoilType == CoolingCoil) {
-                if (!DataGlobals::AnyEnergyManagementSystemInModel) {
+                if (!DataGlobals::AnyEnergyManagementSystemInModel && DataLoopNode::Node(this->CoolCoilOutletNodeNum).HumRatMax == DataLoopNode::SensedNodeFlagValue) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/UnitarySystem.cc:1372]: 👍

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
@rraustad can you resolve the conflicts in the interface manager?  Once that is cleaned up I'll let CI do a full pass today and this could merge in soon.

File: src/EnergyPlus/HVACInterfaceManager.cc (hunk 2 above)
Comment (rraustad) [reply src/EnergyPlus/HVACInterfaceManager.cc]: Someone else also made a change here to protect divide by 0. I'm just going with that other change to pull this out of this PR.

**Comment by rraustad** [comment]
#7431 (SET) diff's crept in on Windows CI. No need to run again, this is ripe for merge.

**Pull request merged**